crates/p2p_proto/proto/sync.proto

```diff
@@ -28,42 +30,32 @@ enum Direction {
   BACKWARD = 1;
 }
 
+// Retrieve at most `count` block headers staring at `start_block`
+// and then moving to the next or previous blocks depending on `direction`.
+// Any single reply must not exceed `size_limit`.
 message GetBlockHeaders {
-  starknet.common.FieldElement start_block =
-      2;            // block hash of the starting block
-  uint64 count = 3; // how many block, at most, to retrieve from this requested
-                    // starting point
-  uint64 size_limit = 4; // limit on size of returned messages.
-
-  Direction direction = 5;
+  uint64 start_block = 1;
```

Comment:
How could we ask for blocks ahead of our local head by hash? It didn't make sense to me unless we solely depended on header propagation to provide this data... but what we receive via gossipsub doesn't necessarily allow us to build a continuous sequence of block hashes to pick from if we wanted to start at a particular block number.

Reply:
You can get a block hash from:

1. L1 (safest)
2. L2 gossip (safe if you can verify consensus)
3. Randomly asking `N` peers for the latest block hash

This makes sense if you are not an archive node. You don't need all of history, so you just start at some latest point, and ask for the previous `N` blocks, and then start operating already.

Reply:
Is there a reason it matters? Sure we might not need use it right now, but in general the thing is just: give me this range of blocks.. whether the caller has a block hash or a number doesn't really matter? Nor does it matter where they got it from? How we use this in our sync is up to our sync algorithm.

Currently its likely just give me block numbers `i:i+1000` or something. But maybe we want to be more security conscious, so we first get the latest L1 block hash or number, and then query backwards from there.

Only supporting block numbers is fine for now. I imagine the block hash thing would be more useful for querying potential forks. But we don't really need those now.